src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -748,6 +768,83 @@ public int lastIndexOf(Object o) {
         }
     }
 
+    @jdk.internal.ValueBased
+    static final class StableList<E> extends AbstractImmutableList<E> {
```

Comment:
Is there significant reuse gained by putting StableList in ImmutableCollection? The back-and-forth between here and SV through SharedSecrets is a little awkward.

Reply:
This allows reuse of `AbstractImmutableList` with list iterators, sub lists and more.

Reply:
Using the regular AbstractList as a base would also get you implementations of those.

Reply:
`AbstractList` has non-final fields, which makes it not suitable for `@ValueBased`.